timer.R

```diff
@@ -0,0 +1,30 @@
+library(tictoc)
+
+# Time the original script
+tic("Original")
+source("scripts/1_hw4_solution_original.R")
+toc()
+
+# Time the script with parallel computing
+tic("Parallel computing")
+source("scripts/2_hw4_solution_parallel_computing.R")
+toc()
+
+# Time the script with rewritten MTweedieTests function to split the 
+# M simulations in more than one core
+tic("Multiple cores")
+source("scripts/3_hw4_solution_multiple_cores.R")
+toc()
+
+
+############ Results ############ 
+# Original: 62.63 sec
+# Parallel: 50.61 sec
+# Multiple cores: 62.16 sec
+
+# The multiple cores script (62.16 sec) might be almost as slow as the original 
+# one because of one of the two reasons:
+# 1. Communication overhead: Splitting simulations across multiple cores may 
+# introduce communication overhead, slowing down the process.
+# 2. Non-parallelizable sections: Some parts of the script can't be parallelized
+# and must be executed sequentially, limiting potential speedup.
```

Comment:
The last one is supposed to be the quickest, and are the quickest on my computer. The reason that the last one is quicker than the second one is because the second one parallelizes the rows in df. Because there are only 5 rows, having more than 5 cores will not make it quicker. The last one parallizes M, and can therefore utilize more cores in the computation. For example, if you use 10 cores, each core will run M/10 simulations each.